python/nav/web/business/reportengine.py

```diff
@@ -0,0 +1,125 @@
+                      period, report_type)
+        report = build_report(period, report_type)
+        subscriptions = ReportSubscription.objects.filter(
+            period=period, report_type=report_type)
+        for subscription in subscriptions:
+            send_report(report, subscription.address.address)
+        _logger.info('%s %s availability: Sent %s reports',
+                     period, report_type, subscriptions.count())
+
+
+def send_report(report, to_address):
+    """Sends a single email report"""
+    _logger.debug('Sending report to %s', to_address)
+    send_mail(
+        report.subject,
+        report.text_message,
+        'noreply@example.com',
+        [to_address],
+        html_message=report.html_message
+    )
+
+
+def build_report(period, report_type):
+    """Builds a report for a given subscription period"""
+    context = build_context(period, report_type)
+    html_message = render_to_string('business/email.html', context)
+    text_message = render_to_string('business/email.txt', context)
+    return Report(get_email_subject(period, report_type),
+                  period, text_message, html_message)
+
+
+def build_context(period, report_type):
+    """Builds a context for the given subscription"""
+    midnight = date.today()
+    start, end = get_last_interval(midnight, period)
+    lookup = {
+        ReportSubscription.DEVICE: utils.get_netbox_records,
+        ReportSubscription.LINK: utils.get_interface_records,
+    }
+    records = lookup[report_type](start, end)
+    return {
+        'start': start,
+        'end': end,
+        'today': midnight,
+        'records': records
+    }
+
+
+def get_last_interval(sometime, period):
+    """Gets the start and end dates for the previous period
+
+    This means that if `sometime` is a datetime object from the current month
+    and `period` indicates month, then this function will return two
+    datetime-objects - the first one is the first day of _previous_ month and
+    the second one is the first day of the _current_ month.
+
+    :returns: start and end date objects
+    :rtype: list[datetime.datetime]
+    """
+    if period == ReportSubscription.MONTH:
+        first_day = sometime.replace(day=1)
+    elif period == ReportSubscription.WEEK:
+        first_day = sometime - timedelta(days=sometime.weekday())
+    else:
+        first_day = sometime  # Daily
+
+    last_period = first_day - timedelta(days=1)
+    start, end = utils.get_interval(last_period, period)
+    return convert_to_datetime(start, end)
+
+
+def convert_to_datetime(*dates):
+    """Converts date or datetime objects to date"""
+    return [datetime(d.year, d.month, d.day) for d in dates]
+
+
+def get_email_subject(period, report_type):
+    """Creates an email subject for the report"""
+    return "{} {} report from NAV".format(
+        ReportSubscription.get_period_description(period),
+        ReportSubscription.get_type_description(report_type)
+    ).title()
```

Comment:
This causes `NAV` to become `Nav` in the subject line, which looks weird.

Also, maybe the e-mail should include a link back to the NAV server, or at least a mention of the NAV server's name (just using NAV may be confusing for the weirdos that like to run multiple NAV's - like ourselves ;) )